src/config.c

```diff
@@ -754,6 +754,104 @@ static int cgroup_config_mount_fs(void)
 	return error;
 }
 
+static int convert_controller_versions(struct cgroup *cgroup)
+{
+	enum cg_version_t current_version, convert_version;
+	struct cgroup *cgrp_cpy, *convert_cgrp = NULL;
+	struct cgroup_controller *cgc;
+	int i, j, ret;
+
+	if (!cgroup)
+		return ECGINVAL;
+
+	/* No controllers added */
+	if (cgroup->index == 0)
+		return ECGINVAL;
+
+	cgrp_cpy = cgroup_new_cgroup(cgroup->name);
+	if (!cgrp_cpy) {
+		cgroup_err("Failed to create cgroup %s\n", cgroup->name);
+		return ECGROUPNOTCREATED;
+	}
+
+	/* Copy the non-controllers details */
+	cgrp_cpy->index = 0;
+	cgrp_cpy->task_fperm = cgroup->task_fperm;
```

Comment:
Would it make sense to call `cgroup_set_permissions()` here?

Reply:
Agreed, will use `cgroup_set_permissions()`.